### PR TITLE
Incremental emissions for `Publishers.lines()`

### PR DESCRIPTION
**Pull request by NiteshKant**
(This contains changes in #22)

__Problem__

Currently `Publishers.lines()` waits for the `CharSource` to terminate before it emits all the items. This when used with `System.in` (channel mode) never emits any item.

__Modification__

Modify the method to emit every line read.

__Result__

Channel mode works now.

### CODE REVIEW

File: src/main/java/io/reactivesocket/cli/Publishers.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+     * Return a publisher for consuming each line from the passed stream.
+     *
+     * @param inputStream to read.
+     */
+    public static Publisher<Payload> lines(CharSource inputStream) {
+        return Flowable.fromPublisher(spiltInLines(inputStream))
+                       .map(l -> (Payload) new PayloadImpl(l));
+    }
+
+    public static Publisher<String> spiltInLines(CharSource inputStream) {
```
Comment (yschimke) [inline src/main/java/io/reactivesocket/cli/Publishers.java:46]: typo

File: src/main/java/io/reactivesocket/cli/Main.java [hunk 2]
```diff
@@ -130,13 +123,9 @@
     public OutputHandler outputHandler;
     private TransportServer.StartedServer server;
 
-    private Logger retainedLogger;
-
     public void run() throws IOException, URISyntaxException, InterruptedException {
         System.setProperty("org.slf4j.simpleLogger.defaultLogLevel", debug ? "debug" : "warn");
 
-        retainedLogger = LoggerFactory.getLogger("");
```
Comment (yschimke) [inline src/main/java/io/reactivesocket/cli/Main.java:138]: without this --debug doesn't work reliably.  Can we put this back.

**Comment by yschimke** [comment]
Thanks for fixing.  I'll put back the logger after confirming its still needed and fix teh typo

**Comment by NiteshKant** [comment]
@yschimke 

>>> fix teh typo

How ironic 😉

**Comment by yschimke** [comment]
I did some quick testing with "-i @/usr/share/dict/words" but I only got a single result "A".  Is there anything still missing?

**Comment by NiteshKant** [comment]
>>>I did some quick testing with "-i @/usr/share/dict/words" but I only got a single result "A". Is there anything still missing?

Did you try with an echo channel implementation? If so, then most likely, it is due to the lack of request-n sent to the server. I tested `Publishers` with the dictionary and it emits all words.

**Pull request merged**